thewalrus/symplectic.py

```diff
@@ -361,3 +363,53 @@ def autonne(A, rtol=1e-05, atol=1e-08, svd_order=True):
     if svd_order:
         return (vals[n : 2 * n])[::-1], U[:, ::-1]
     return vals[n : 2 * n], U
+
+def xxpp_to_xpxp(S):
+    """Permutes the entries of the input from xxpp ordering to xpxp ordering.
+
+    Args:
+        S (array): input even dimensional square matrix or array
+
+    Returns:
+        (array): permuted matrix or array
+    """
+    shape = S.shape
+    n = shape[0]
+
+    if n % 2 != 0:
+        raise ValueError("The input array is not even dimensional")
```

Comment:
I'm reading this as that "the input array is not dimensional", with the "even" just being an intensifier. 😆

Reply:
your array is not even dimensional!!!!

Reply:
ah... dashes! Thanks for catching that.